diff --git a/imgui/funcs.go b/imgui/funcs.go
index 98bb1d24..706f4d6d 100644
--- a/imgui/funcs.go
+++ b/imgui/funcs.go
@@ -11,6 +11,7 @@ import (
 	"unsafe"
 
 	"github.com/AllenDang/cimgui-go/internal"
+	"github.com/AllenDang/cimgui-go/utils/datatypes"
 	"github.com/AllenDang/cimgui-go/utils/vectors"
 )
 
@@ -23089,13 +23090,13 @@ func (self *Context) TooltipPreviousWindow() *Window {
 	return NewWindowFromC(C.wrap_ImGuiContext_GetTooltipPreviousWindow(internal.ReinterpretCast[*C.ImGuiContext](selfArg)))
 }
 
-func (self *Context) ClipboardHandlerData() vectors.Vector[int8] {
+func (self *Context) ClipboardHandlerData() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Size, C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Capacity, (*int8)(C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Size, C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiContext_GetClipboardHandlerData(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Data))
 }
 
 func (self *Context) MenusIdSubmittedThisFrame() vectors.Vector[ID] {
@@ -23674,13 +23675,13 @@ func (self *Context) WantTextInputNextFrame() int32 {
 	return int32(C.wrap_ImGuiContext_GetWantTextInputNextFrame(internal.ReinterpretCast[*C.ImGuiContext](selfArg)))
 }
 
-func (self *Context) TempBuffer() vectors.Vector[int8] {
+func (self *Context) TempBuffer() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Size, C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Capacity, (*int8)(C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Size, C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiContext_GetTempBuffer(internal.ReinterpretCast[*C.ImGuiContext](selfArg)).Data))
 }
 
 func (self *Context) TempKeychordName() [64]rune {
@@ -27938,13 +27939,13 @@ func (self *InputTextDeactivatedState) ID() ID {
 	return *NewIDFromC(func() *C.ImGuiID { result := result; return &result }())
 }
 
-func (self *InputTextDeactivatedState) TextA() vectors.Vector[int8] {
+func (self *InputTextDeactivatedState) TextA() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Size, C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Capacity, (*int8)(C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Size, C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiInputTextDeactivatedState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextDeactivatedState](selfArg)).Data))
 }
 
 func (self InputTextState) SetCtx(v *Context) {
@@ -28150,31 +28151,31 @@ func (self *InputTextState) TextSrc() string {
 	return C.GoString(C.wrap_ImGuiInputTextState_GetTextSrc(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)))
 }
 
-func (self *InputTextState) TextA() vectors.Vector[int8] {
+func (self *InputTextState) TextA() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*int8)(C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiInputTextState_GetTextA(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
 }
 
-func (self *InputTextState) TextToRevertTo() vectors.Vector[int8] {
+func (self *InputTextState) TextToRevertTo() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*int8)(C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiInputTextState_GetTextToRevertTo(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
 }
 
-func (self *InputTextState) CallbackTextBackup() vectors.Vector[int8] {
+func (self *InputTextState) CallbackTextBackup() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*int8)(C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Size, C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiInputTextState_GetCallbackTextBackup(internal.ReinterpretCast[*C.ImGuiInputTextState](selfArg)).Data))
 }
 
 func (self *InputTextState) BufCapacity() int32 {
@@ -37405,13 +37406,13 @@ func (self TextBuffer) SetBuf(v vectors.Vector[int8]) {
 	C.wrap_ImGuiTextBuffer_SetBuf(selfArg, *vVecArg)
 }
 
-func (self *TextBuffer) Buf() vectors.Vector[int8] {
+func (self *TextBuffer) Buf() vectors.Vector[datatypes.Char] {
 	selfArg, selfFin := self.Handle()
 
 	defer func() {
 		selfFin()
 	}()
-	return vectors.NewVectorFromC(C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Size, C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Capacity, (*int8)(C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Data))
+	return vectors.NewVectorFromC(C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Size, C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Capacity, (*datatypes.Char)(C.wrap_ImGuiTextBuffer_GetBuf(internal.ReinterpretCast[*C.ImGuiTextBuffer](selfArg)).Data))
 }
 
 func (self TextFilter) SetInputBuf(v *[256]rune) {
diff --git a/internal/type_wrapper.go b/internal/type_wrapper.go
index f2129bc7..5026d548 100644
--- a/internal/type_wrapper.go
+++ b/internal/type_wrapper.go
@@ -8,6 +8,12 @@ import (
 	"unsafe"
 )
 
+// Char represents C.char. On amd64 it is equivalent to int8, on arm64 it is uint8.
+// See https://github.com/AllenDang/cimgui-go/issues/140
+type Char interface {
+	~int8 | ~uint8
+}
+
 // ReinterpretCast acts exactly like C++'s reinterpret_cast.
 // Intendedd use is to convert packageA.C.MyType to packageB.C.MyType.
 // make sure your types are identical C types before using it.
@@ -19,7 +25,7 @@ func ReinterpretCast[RET, SRC any](src SRC) RET {
 
 // WrapString converts Go string to C char*
 // Default value of RET is C.char
-func WrapString[RET ~int8](value string) (wrapped *RET, finisher func()) {
+func WrapString[RET ~Char](value string) (wrapped *RET, finisher func()) {
 	wrapped = ReinterpretCast[*RET](C.CString(value))
 	finisher = func() { C.free(unsafe.Pointer(wrapped)) } // nolint: gas
 	return
@@ -27,7 +33,7 @@ func WrapString[RET ~int8](value string) (wrapped *RET, finisher func()) {
 
 // WrapStringList converts Go string slice to C char**
 // Default value of RET is C.char
-func WrapStringList[RET ~int8](value []string) (wrapped **RET, finisher func()) {
+func WrapStringList[RET ~Char](value []string) (wrapped **RET, finisher func()) {
 	if len(value) == 0 {
 		return nil, func() {}
 	}
diff --git a/utils/datatypes/char_signed.go b/utils/datatypes/char_signed.go
new file mode 100644
index 00000000..45fe6d5b
--- /dev/null
+++ b/utils/datatypes/char_signed.go
@@ -0,0 +1,7 @@
+//go:build amd64 || (arm64 && darwin)
+
+package datatypes
+
+// Char represents underlying type of C.char. On amd64 it is equivalent to int8, on arm64 it is uint8.
+// See https://github.com/AllenDang/cimgui-go/issues/140
+type Char int8
diff --git a/utils/datatypes/char_unsigned.go b/utils/datatypes/char_unsigned.go
new file mode 100644
index 00000000..3bb2f1c6
--- /dev/null
+++ b/utils/datatypes/char_unsigned.go
@@ -0,0 +1,7 @@
+//go:build arm64 && !darwin
+
+package datatypes
+
+// Char represents underlying type of C.char. On amd64 it is equivalent to int8, on arm64 it is uint8.
+// See https://github.com/AllenDang/cimgui-go/issues/140
+type Char uint8
