diff --git a/tts/tts.go b/tts/tts.go
index c7c3c3bf..fb1db9eb 100644
--- a/tts/tts.go
+++ b/tts/tts.go
@@ -5,12 +5,12 @@
 package tts
 
 /*
-#cgo CFLAGS: -I${SRCDIR}/../sherpa-onnx/sherpa-onnx/c-api
+#cgo CFLAGS: -I/app/include/sherpa-onnx/c-api
 
-#cgo !windows LDFLAGS: ${SRCDIR}/../sherpa-onnx/build_go/lib/libsherpa-onnx-c-api.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libsherpa-onnx-core.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libsherpa-onnx-fstfar.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libsherpa-onnx-fst.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libsherpa-onnx-kaldifst-core.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libkaldi-decoder-core.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libkaldi-native-fbank-core.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libssentencepiece_core.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libpiper_phonemize.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libespeak-ng.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libucd.a ${SRCDIR}/../sherpa-onnx/build_go/lib/libkissfft-float.a -lm
+#cgo !windows LDFLAGS: -lsherpa-onnx-c-api -lsherpa-onnx-core -lsherpa-onnx-fstfar -lsherpa-onnx-fst -lsherpa-onnx-kaldifst-core -lkaldi-decoder-core -lkaldi-native-fbank-core -lssentencepiece_core -lpiper_phonemize -lespeak-ng -lucd -lkissfft-float -lm
 
 #cgo darwin LDFLAGS: ${SRCDIR}/../sherpa-onnx/build_go/_deps/onnxruntime-src/lib/libonnxruntime.a -lstdc++ -framework Foundation -framework Accelerate
-#cgo linux LDFLAGS: ${SRCDIR}/../sherpa-onnx/build_go/_deps/onnxruntime-src/lib/libonnxruntime.a -lstdc++ -lpthread -ldl
+#cgo linux LDFLAGS: -lonnxruntime -lstdc++ -lpthread -ldl
 #cgo windows LDFLAGS: -L${SRCDIR}/../sherpa-onnx/build_go/lib -lsherpa-onnx-c-api -L${SRCDIR}/../sherpa-onnx/build_go/_deps/onnxruntime-src/lib -lonnxruntime -static-libstdc++ -static-libgcc
 
 #include <c-api.h>
